test/tests/functional/pbs_multi_sched.py

```diff
@@ -647,6 +647,268 @@ def test_multi_fairshare(self):
+                                                 202, 'grp2', 10)
+        # Set scheduler iteration
+        sc_iter = {'scheduler_iteration': 7}
+        self.server.manager(MGR_CMD_SET, SCHED, sc_iter, id='default')
+        self.server.manager(MGR_CMD_SET, SCHED, sc_iter, id='sc1')
+        self.server.manager(MGR_CMD_SET, SCHED, sc_iter, id='sc2')
+        # Update scheduler config file
+        formula = 'ceil(fabs(-ncpus*(foo/100.00)*sqrt(100)))'
+        sc_config = {'fair_share': 'True', 'fairshare_decay_time': '24:00:00',
+                     'fairshare_decay_factor': 0.5,
+                     'fairshare_usage_res': formula}
+        self.scheds['default'].set_sched_config(sc_config)
+        self.scheds['sc1'].set_sched_config(sc_config)
+        self.scheds['sc2'].set_sched_config(sc_config)
+        # Set scheduling false
+        sc_scheduling = {'scheduling': 'False'}
+        self.server.manager(MGR_CMD_SET, SCHED, sc_scheduling, id='default')
+        self.server.manager(MGR_CMD_SET, SCHED, sc_scheduling, id='sc1')
+        self.server.manager(MGR_CMD_SET, SCHED, sc_scheduling, id='sc2')
+        # Set resources to node
+        resc = {'resources_available.ncpus': 1,
+                'resources_available.foo': 5000}
+        self.server.manager(MGR_CMD_SET, NODE, resc, self.mom.shortname)
+        self.server.manager(MGR_CMD_SET, NODE, resc, id='vnode[0]')
+        self.server.manager(MGR_CMD_SET, NODE, resc, id='vnode[1]')
+        self.server.manager(MGR_CMD_SET, NODE,
+                            {'resources_available.ncpus': 0},
+                            id='vnode[4]')
+
+        # submit jobs to default scheduler
+        def_attr = {'Resource_List.select': '1:ncpus=1:foo=20',
+                    'Resource_List.walltime': 4}
+        def_J1 = Job(TEST_USER1, attrs=def_attr)
+        def_jid1 = self.server.submit(def_J1)
+        def_J2 = Job(TEST_USER2, attrs=def_attr)
+        def_jid2 = self.server.submit(def_J2)
+        def_J3 = Job(TEST_USER3, attrs=def_attr)
+        def_jid3 = self.server.submit(def_J3)
+        self.server.manager(MGR_CMD_SET, SCHED, {'scheduling': 'True'},
+                            id='default')
+        # pbsuser1 job will run and other two will be queued
+        self.server.expect(JOB, {'job_state': 'R'}, id=def_jid1)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=def_jid3)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=def_jid2)
+        # pbsuser3 job will run after pbsuser1
+        self.server.expect(JOB, {'job_state': 'R'}, id=def_jid3)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=def_jid2)
+        # pbsuser2 job will run in the end
+        self.server.expect(JOB, {'job_state': 'R'}, id=def_jid2)
+        self.server.expect(JOB, {'job_state': 'F'}, id=def_jid2, extend='x')
+        # query fairshare and check usage
+        USER1 = str(TEST_USER1)
+        USER2 = str(TEST_USER2)
+        USER3 = str(TEST_USER3)
+        USER4 = str(TEST_USER4)
```

Comment:
Either use these throughout your test plan or just call query_fairshare() with str().  You do a combination of both.  My vote would be to use str() in query_fairshare() itself since it'd be easier to read.  You call Job() with one of the users.  This way you're using the same username.  Either that or update query_fairshare() in the framework to take a PbsUser object.  It isn't hard, just call str() on it right before its use.  That is a bit out of scope for this PR though.